crates/bin/src/dump.rs

```diff
@@ -71,16 +71,20 @@ async fn dump_range(
 ) -> eyre::Result<()> {
     let mut set = tokio::task::JoinSet::new();
 
+    let multi_progress_bar = MultiProgress::new();
+
     for block in range {
         let provider = provider.clone();
         let out_dir = out_dir.clone();
+        let progress_bar = multi_progress_bar.add(ProgressBar::new_spinner());
         set.spawn(async move {
```

Comment:
if we downloading a large range, current would spawn all task concurrently. though we have request limit in provider level, i'm thinking should we limit this, otherwise it will overflow our terminal?